pandas/core/arrays/floating.py

```diff
@@ -257,6 +257,15 @@ def __init__(self, values: np.ndarray, mask: np.ndarray, copy: bool = False):
             )
         super().__init__(values, mask, copy=copy)
 
+    def __neg__(self):
+        return type(self)(-self._data, self._mask)
```

Comment:
should return a copy of the mask, xref #39943

should probably move the integer implementations (after fixed https://github.com/pandas-dev/pandas/issues/39943#issuecomment-783383557) to the base clase.